gap/examples.gi

```diff
@@ -369,6 +369,183 @@ GeneralisedPetersenGraphCons);
 InstallMethod(GeneralisedPetersenGraph, "for integer, integer", [IsInt, IsInt],
 {n, k} -> GeneralisedPetersenGraphCons(IsImmutableDigraph, n, k));
 
+InstallMethod(BishopsGraphCons,
+"for IsMutableDigraph, a string and two positive integers",
+[IsMutableDigraph, IsString, IsPosInt, IsPosInt],
+function(filt, color, m, n)
+  local D, D1, D2, i, j, v, vertices, map;
+
+  if not (color = "dark" or color = "light") then
+    ErrorNoReturn(
+    "the argument <color> must be either \"dark\" or \"light\".");
+  fi;
+
+  vertices := EuclideanQuotient(m * n, 2);
+
+  if IsOddInt(m * n) and color = "dark" then
+    vertices := vertices + 1;
+  fi;
+
+  D1 := EmptyDigraph(IsMutableDigraph, vertices);
+  D2 := EmptyDigraph(IsMutableDigraph, vertices);
+
+  map := function(a)
+      return EuclideanQuotient(a + 1, 2);
+  end;
+
+  for i in [1 .. (m - 1)] do
+    for j in [1 .. (n - 1)] do
+      if IsEvenInt(i + j) and color = "dark"
+          or IsOddInt(i + j) and color = "light" then
+        v := (i - 1) * n + j;
+        DigraphAddEdge(D1, [map(v), map(v + n + 1)]);
+      elif IsEvenInt(i + j) and color = "light"
+          or IsOddInt(i + j) and color = "dark" then
+        v := (i - 1) * n + j + 1;
+        DigraphAddEdge(D2, [map(v), map(v + n - 1)]);
+      fi;
+    od;
+  od;
+  DigraphTransitiveClosure(D1);
+  DigraphTransitiveClosure(D2);
+  D := DigraphEdgeUnion(D1, D2);
+  DigraphSymmetricClosure(D);
+  return D;
+end);
+
+InstallMethod(BishopsGraphCons,
+"for IsImmutableDigraph, a string and two positive integers",
+[IsImmutableDigraph, IsString, IsPosInt, IsPosInt],
+function(filt, color, m, n)
+  local D;
+  D := MakeImmutable(BishopsGraphCons(IsMutableDigraph, color, m, n));
+  SetIsMultiDigraph(D, false);
+  SetIsSymmetricDigraph(D, true);
+  SetIsConnectedDigraph(D, true);
```

Comment:
This is not necessarily true:
```
gap> D := BishopsGraph("dark", 1, 6);
<immutable empty digraph with 3 vertices>
gap> IsConnectedDigraph(D);
true
gap> DigraphNrConnectedComponents(D);
3
```

Reply:
Right, thanks for pointing that out! I'd gotten a bit confused with the definition of a connected graph.